app/controllers/concerns/idv/document_capture_concern.rb

```diff
@@ -81,6 +81,21 @@ def redirect_to_correct_vendor(vendor, in_hybrid_mobile)
       redirect_to correct_path
     end
 
+    def capture_app_analytics(timer)
+      document_request_body = JSON.parse(@document_request.body, symbolize_names: true)[:config]
+      log_extras = {
+        vendor: 'Socure',
+        reference_id: @document_response.to_h[:referenceId],
+        vendor_request_time_in_ms: timer.results['vendor_request'],
+        success: !@url.nil?,
+        document_type: document_request_body[:documentType],
+      }
+      @document_response.to_h.merge(log_extras).merge(analytics_arguments).
+        merge(document_request_body).except(
+          :referenceId, :documentType, :extra
+        ).compact
```

Comment:
```suggestion
      document_request_body = JSON.parse(@document_request.body, symbolize_names: true)[:config]
      log_extras = {
        reference_id: @document_response.to_h[:referenceId],
        vendor_request_time_in_ms: timer.results['vendor_request'],
        success: url.present?,
        document_type: document_request_body[:documentType],
      }
      log_extras.merge(analytics_arguments).
        merge(document_request_body).except(
          :documentType, # requested document type
        ).merge(response_body: @document_response.to_h)
```

extra is helpful to keep

i think it could be helpful here to put the response in a key ... allowing us to group the response details unaltered
and easily distinguish elements  (ie: like "language" requested vs responded)